Sources/Basic/TemporaryFile.swift

```diff
@@ -0,0 +1,172 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+ -------------------------------------------------------------------------
+
+ This file provides API to create and use temporary files and directories.
+*/
+
+import libc
+import func POSIX.getenv
+import class Foundation.NSFileHandle
+import var Foundation.NSUTF8StringEncoding
+import func Utility.fputs // FIXME: we should move file related helper methods to Basic.
+
+public enum TempFileError: ErrorProtocol {
+    /// Could not create a unique temporary filename.
+    case couldNotCreateUniqueName
+    /// The last 6 characters of template were not XXXXXX.
+    case badTemplateName
+    /// Some error thrown defined by posix's open().
+    // FIXME: This should be factored out into a open error enum.
+    case other(Int32)
+    /// Error while decoding unicode.
+    case unicodeDecodingError // FIXME: Already present in Utility.Error, Remove when file is shifted to basic.
+}
+
+private extension TempFileError {
+    init(errno: Int32) {
+        switch errno {
+        case libc.EEXIST:
+            self = .couldNotCreateUniqueName
+        case libc.EINVAL:
+            self = .badTemplateName
+        default:
+            self = .other(errno)
+        }
+    }
+}
+
+/// This class is basically a wrapper over posix's mkstemps() function to creates disposable files.
+/// The file is deleted as soon as the object of this class is deallocated.
+public final class TemporaryFile {
+    /// If specified during init, the temporary file name begins with this prefix.
+    public let prefix: String
+
+    /// If specified during init, the temporary file name ends with this suffix.
+    public let suffix: String
+
+    /// The directory in which the temporary file is created.
+    public let dir: String
+
+    /// The full path of the temporary file. It is valid till the TemporaryFile object is alive.
```

Comment:
> It is valid till the TemporaryFile object is alive 

This goes without saying, since the client can only access it as long as the `TemporaryFile` is alive. I suspect what you meant here is that the file on disk exists as long as the `TemporaryFile` exists, but that is a somewhat misleading thing for a client to care about, since, from a security/correctness perspective, there is no guarantee that that is true. I would just drop this comment and assume clients understand how temporary files work.

If the goal here is to document the "delete on deinit" behavior, then I would just worry about documenting that in the class docs and the init docs (both of which you already did).